### PR TITLE
Invoke-DbaQuery, added regr tests and TVP examples

### PR DESCRIPTION
**Pull request by niphlod**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [x] Unit test is included
 - [x] Documentation
 - [ ] Build system

### Purpose
Fixed a mishap for people used to hashtables for params (also, saved from the obligation of using -ParameterName in that case)
Added an example for TVP

### Approach
When hashtables are used, name is set in stone, so we use that to set (or alter) the ParameterName

### Commands to test
Added regr tests at the bottom.

NB: quick and dirty edit from Notepad++, may the gods of appveyor have mercy on me

### CODE REVIEW

**Comment by niphlod** [comment]
groan, start-dbamigration is failing on me ;-(

**Comment by potatoqualitee** [comment]
Let's see how your new retry helps!

File: internal/functions/Invoke-DbaAsync.ps1 [hunk 1]
```diff
@@ -187,6 +187,9 @@ function Invoke-DbaAsync {
                     ($SqlParameter | Select-Object -First 1).GetEnumerator() | ForEach-Object {
                         if ($null -ne $_.Value) {
                             if (($_.Value -is [Microsoft.Data.SqlClient.SqlParameter])) {
+                                if ($_.Value.ParameterName -ne $_.Key) {
+                                    $_.Value.ParameterName = $_.Key
```
Comment (potatoqualitee) [inline internal/functions/Invoke-DbaAsync.ps1:191]: OHHH! Good catch!

**Comment by niphlod** [comment]
@potatoqualitee : functionality wise, yes, it works! https://ci.appveyor.com/project/sqlcollaborative/dbatools/builds/40015890/job/4gvabb0ehs964a94/tests (see Test-DbaMaxDop) .

Remains to fix the testname reported, first loop is **Test-DbaMaxDop.Tests.ps1**, second loop is **C:\github\dbatools\tests\Test-DbaMaxDop.Tests.ps1.Name, attempt #2** but all things considered, it seems to work fine.

**Pull request merged**